contrib/babelfishpg_tsql/sql/sys_views.sql

```diff
@@ -1960,17 +1960,19 @@ GRANT SELECT ON sys.endpoints TO PUBLIC;
 create or replace view sys.index_columns
 as
 select i.indrelid::integer as object_id
-  , CAST(CASE WHEN i.indisclustered THEN 1 ELSE 1+row_number() OVER(PARTITION BY c.oid) END AS INTEGER) AS index_id
+  , imap.index_id
   , a.attrelid::integer as index_column_id
-  , a.attnum::integer as column_id
+  , col.attnum::integer as column_id
   , a.attnum::sys.tinyint as key_ordinal
   , 0::sys.tinyint as partition_ordinal
   , 0::sys.bit as is_descending_key
   , 1::sys.bit as is_included_column
 from pg_index as i
 inner join pg_catalog.pg_attribute a on i.indexrelid = a.attrelid
+inner join pg_catalog.pg_attribute col on col.attname = a.attname and col.attrelid = i.indrelid
```

Comment:
This condition col.attname = a.attname and col.attrelid = i.indrelid will ensure that col.attnum will be same as a.attnum and col.relid will be same as a.relid so I am not convinced why do we need this change.

Reply:
column_id will return the position of the column from the table definition,
key_ordinal will return the position of the column from the index definiton.

Reply:
```
create table t4 (c1 int, c2 int, c3 int, c4 int, c5 int, primary key clustered (c3, c5))
select 't3' as table_name, column_id, key_ordinal from sys.index_columns where object_id =  object_id('t5')
table_name column_id   key_ordinal
---------- ----------- -----------
t3                   3           1
t3                   5           2
```